torch/optim/nadam.py

```diff
@@ -242,10 +248,21 @@ def _single_tensor_nadam(params: List[Tensor],
         # decay the first and second moment running average coefficient
         exp_avg.mul_(beta1).add_(grad, alpha=1 - beta1)
         exp_avg_sq.mul_(beta2).addcmul_(grad, grad, value=1 - beta2)
-
-        denom = exp_avg_sq.div(bias_correction2).sqrt().add_(eps)
-        param.addcdiv_(grad, denom, value=-lr * (1. - mu) / (1. - mu_product.item()))
-        param.addcdiv_(exp_avg, denom, value=-lr * mu_next / (1. - mu_product_next.item()))
+        denom = exp_avg_sq.div(bias_correction2).sqrt()
+        if differentiable:
+            denom = denom.add(eps)
+            # Make autograd track the operations
```

Comment:
nit: could you append: "by updating the grad and exp_avg directly and not using the scalar "value" argument of addcdiv.

btw on GPU, it might be better to do this as the item() would cause a sync cc @ptrblck if someone on your end has time to check this out?